packages/web-app-ocm/src/composables/useInviteTokens/useInviteTokens.ts

```diff
@@ -0,0 +1,105 @@
+import { ref } from 'vue'
+import { useClientService, useConfigStore } from '@ownclouders/web-pkg'
+import { inviteListSchema, inviteSchema } from '../../schemas'
+
+type Token = {
+  id: string
+  token: string
+  link?: string
+  expiration?: Date
+  expirationSeconds?: number
+  description?: string
+}
+
+const tokensList = ref<Token[]>([])
+const lastCreatedToken = ref('')
+let clientService
+
+const toDateTime = (secs: number) => {
+  const d = new Date(Date.UTC(1970, 0, 1))
+  d.setUTCSeconds(secs)
+  return d
+}
+
+function encodeInviteToken(token: string) {
+  const configStore = useConfigStore()
+  const url = new URL(configStore.serverUrl)
+  return btoa(`${token}@${url.host}`)
+}
+
+async function fetchTokens() {
+  const url = '/sciencemesh/list-invite'
+
+  try {
+    const { data } = await clientService.httpAuthenticated.get(url, {
+      schema: inviteListSchema
+    })
+    data.forEach((t) => {
+      tokensList.value.push({
+        id: t.token,
+        token: t.token,
+        ...(t.expiration && {
+          expiration: toDateTime(t.expiration)
+        }),
+        ...(t.expiration && {
+          expirationSeconds: t.expiration
+        }),
+        ...(t.description && { description: t.description })
+      })
+    })
+  } catch (error) {
+    throw new Error(error)
+  }
+}
+
+async function generateToken(description: string) {
```

Comment:
Why not have this inside the `useInviteTokens` composable so that you do not need to have the top level variable?